R/epi_keys.R

```diff
@@ -29,7 +29,7 @@ epi_keys.recipe <- function(x) {
 epi_keys_mold <- function(mold) {
   keys <- c("time_value", "geo_value", "key")
   molded_names <- names(mold$extras$roles)
-  mold_keys <- purrr::map_chr(mold$extras$roles[molded_names %in% keys], names)
+  mold_keys <- map_chr(mold$extras$roles[molded_names %in% keys], names)
```

Comment:
Will this work when there are multiple `other_key`s? I know we tried it in the meeting last week but I can't remember. The fix I had was to just use `map` and then `unlist(unname(mold_keys))`.

Reply:
It should. All `other_keys` get assigned the role "key". So this is a place where screwing up the `epi_df` metadata could break epipredict.

Reply:
I pulled your branch and ran this test that I wrote for `test-epi_keys`, and it doesn't pass. Maybe I'm missing something? @dajmcdon 

```
test_that("epi_keys_mold extracts additional keys when they are present", {
  my_data <- case_death_rate_subset %>%
    sample_n(6) %>%
    tsibble::as_tsibble() %>% # add 2 extra keys
    mutate(state = rep("MA", 6), pol = rep("blue", 6)) %>% 
    as_epi_df(additional_metadata = list(other_keys=c("state", "pol"))) # <-- metadata constructed here

  my_recipe <- epi_recipe(my_data) %>%
    step_epi_ahead(death_rate , ahead = 7) %>%
    step_epi_naomit()

  my_workflow <- epi_workflow(my_recipe, linear_reg()) %>% fit(my_data)

  expect_equal(
    epi_keys_mold(my_workflow$pre$mold), 
    c("time_value", "geo_value", "state", "pol"))
})
```

Error msg:
```
Error in `vapply(.x, .f, .mold, ..., USE.NAMES = FALSE)`: values must be length 1,
 but FUN(X[[3]]) result is length 2
Backtrace:
 1. testthat::expect_equal(...)
 4. epipredict::epi_keys_mold(my_workflow$pre$mold)
 5. epipredict::map_chr(...)
      at epipredict/R/epi_keys.R:32:2
 6. epipredict::.rlang_purrr_map_mold(.x, .f, character(1), ...)
      at epipredict/R/compat-purrr.R:32:2
 7. base::vapply(.x, .f, .mold, ..., USE.NAMES = FALSE)
      at epipredict/R/compat-purrr.R:49:2

Error in reporter$stop_if_needed() : Test failed
Backtrace:
1: stop(error)
2: (function (envir) 
3: test_that("epi_keys_mold extracts additional keys when they are present", 
```

Reply:
Nuts. Good catch. Should be fixed now. I added your code above to the test.

Reply:
Bugger. This fails tests. The reason being that `tsibble` is not imported. I don't want it for one test. And I'm having trouble adding the additional metadata without converting to tsibble. Can you fix it up on your local version?

Reply:
Here, this one doesn't use `tsibble`:
```
test_that("epi_keys_mold extracts additional keys when they are present", {
  my_data <- tibble::tibble(
    geo_value = rep(c("ca", "fl", "pa"), each = 3), 
    time_value = rep(seq(as.Date("2020-06-01"), as.Date("2020-06-03"),
                            by = "day"), length.out = length(geo_value)),
    pol = rep(c("blue", "swing", "swing"), each = 3), # extra key
    state = rep(c("ca", "fl", "pa"), each = 3), # extra key
    value = 1:length(geo_value) + 0.01 * rnorm(length(geo_value))
  ) %>%
  as_epi_df(additional_metadata = list(other_keys = c("state", "pol")))

  my_recipe <- epi_recipe(my_data) %>%
    step_epi_ahead(value , ahead = 7) %>%
    step_epi_naomit()

  my_workflow <- epi_workflow(my_recipe, linear_reg()) %>% fit(my_data)

  expect_setequal(
    epi_keys_mold(my_workflow$pre$mold), 
    c("time_value", "geo_value", "state", "pol"))
})
```

Reply:
Did you want me to push it to your branch myself? I can definitely do that too